### ISSUE TITLE
[feature]: Centralized Translation File for Multi-Language Support

### ISSUE THREAD

**Issue by Ale1x** [makeplane/plane#2028]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

I'd like to propose a feature that allows for the translation of the app into multiple languages by placing all translatable strings into a single file. This would streamline the localization process and make it easier for developers and translators to manage translations.

### Why should this be worked on?

**Simplicity**: Having a centralized file for translations reduces the complexity of managing multiple files or scattered strings throughout the codebase.
**Efficiency**: Translators can focus on a single file, ensuring consistency and reducing the likelihood of missing any strings.
Scalability: As the app grows and more languages are added, it becomes easier to manage and update translations without combing through multiple files.
**Collaboration**: A single file can be easily shared with translation teams or platforms, facilitating collaboration and faster translation turnaround.
**Maintenance**: Future updates or changes to the text can be made in one place, ensuring that all language versions remain synchronized.
**User Experience**: Provides a seamless experience for users across different regions, as all translations would be consistent and up-to-date.

**Comment by rhea0110** [comment]
Hello @Ale1x,

Thank you for the feature request. We already have this in our pipeline, and our team will be working on it in the near future.

**Comment by mateusztylec** [comment]
I will be happy to help with the Polish translation!

**Comment by yasin172** [comment]
Your suggestion is spot on and holds significant importance for modern software development processes. Having all translatable strings in a single file would streamline the process of creating a multilingual app or website. Considering the constant change and globalization in the technology and software sector, such an approach is bound to save time and accelerate workflows.

This concept should definitely be more widely adopted by international developers and coders. Consolidating translations into a single file is not only beneficial for the reasons you pointed out above, but it also provides a sustainable and scalable solution. In short, this suggestion should be popularized and standardized within the industry  @Ale1x @rhea0110

**Comment by vihar** [comment]
Thank you, @Ale1x and @yasin172. We will address this issue shortly and share ideas once it moves out of the backlog. Meanwhile, to facilitate easier communication, I'll be sharing feedback and interest from other users in this thread.

https://github.com/makeplane/plane/issues/2222 - @AkiraVoid's help for translating in English, Chinese, and Japanese.

**Comment by FranziskaSom** [comment]
> Thank you, @Ale1x and @yasin172. We will address this issue shortly and share ideas once it moves out of the backlog. Meanwhile, to facilitate easier communication, I'll be sharing feedback and interest from other users in this thread.
> 
> #2222 - @AkiraVoid's help for translating in English, Chinese, and Japanese.
> 
> #2178 - https://github.com/DenisVASI9's help for Russian/Serbia translation.
> 
> #2178 - [gfviegas](https://github.com/gfviegas)'s help for Portuguese translation.

we are happy to do the german translation ...

**Comment by Ale1x** [comment]
Nice, I would really like to help with the Italian translation

> > Thank you, @Ale1x and @yasin172. We will address this issue shortly and share ideas once it moves out of the backlog. Meanwhile, to facilitate easier communication, I'll be sharing feedback and interest from other users in this thread.
> > #2222 - @AkiraVoid's help for translating in English, Chinese, and Japanese.
> > #2178 - https://github.com/DenisVASI9's help for Russian/Serbia translation.
> > #2178 - [gfviegas](https://github.com/gfviegas)'s help for Portuguese translation.
> 
> we are happy to do the german translation ...

**Comment by arktnld** [comment]
Any news about this? I can't find any option to change the language

**Comment by MBeggiato** [comment]
> > Thank you, @Ale1x and @yasin172. We will address this issue shortly and share ideas once it moves out of the backlog. Meanwhile, to facilitate easier communication, I'll be sharing feedback and interest from other users in this thread.
> > #2222 - @AkiraVoid's help for translating in English, Chinese, and Japanese.
> > #2178 - https://github.com/DenisVASI9's help for Russian/Serbia translation.
> > #2178 - [gfviegas](https://github.com/gfviegas)'s help for Portuguese translation.
> 
> we are happy to do the german translation ...

Same for me, I would love to help to translate into german!

**Comment by Julian-Dumitrascu** [comment]
When somebody places an order with [Sol Language Services](https://github.com/Sol-Global-Management/language-services/issues/1), we can work with any number of files and manage the content in each language in which one wants to do business.

**Comment by emmgfx** [comment]
I could collaborate with the spanish translation :)

**Comment by arktnld** [comment]
I can help with portuguese

**Comment by danielcolgan** [comment]
I could collaborate with the ukrainian translation! +1

**Comment by vihar** [comment]
Support for Chinese language from https://github.com/makeplane/plane/issues/4158.

**Comment by AliDjango** [comment]
I can help with Persian translation, although this requires RTL Support and Persian Calendar support too...

**Comment by ArtMiranda** [comment]
I can collaborate with the (Brazilian) Portuguese as well!

**Comment by rasyidly** [comment]
I can collaborate to Indonesian as well :)

**Comment by Ale1x** [comment]
Waiting for updates @rhea0110

**Comment by elmagnificogi** [comment]
I could collaborate with the chinese translation :)

**Comment by nicobts** [comment]
Hi! I am totally available to collaborate for the Italian translation

**Comment by henriquelatorre** [comment]
I am also interested in contributing with Brazilian Portuguese. I suggest setting up a weblate or crowdin translation server for managing translations outside github

**Comment by zhangyan1022** [comment]
Can I know how long it will take to support Chinese?

**Comment by sriramveeraghanta** [comment]
We are in process of implementing language support across the platform, phase by phase. Once the implementation is done, It is matter to creating new locale files for all the languages. We will keep you posted if there is any update.

**Comment by nicobts** [comment]
@sriramveeraghanta perfect thanks a lot for the answer.
If you need help for the translation of the locales feel free to ask. Happy to support!

**Comment by Gabrielloopes33** [comment]
I can help with Brazilian Portuguese translation

**Comment by vihar** [comment]
folks, language support is now enabled on this repository. read the instructions [here](https://github.com/makeplane/plane/blob/preview/CONTRIBUTING.md#contributing-to-language-support) for contributions. 

cc: @sriramveeraghanta, @AnmolBhatia1001, @prateekshourya29

**Comment by sriramveeraghanta** [comment]
Language support is live in version v0.25.0, Please find the contribution instructions [here](https://github.com/makeplane/plane/blob/preview/CONTRIBUTING.md#contributing-to-language-support). Closing this, please open a new issue for any issues related to language support.
